internal/parser/parser.go

```diff
@@ -30,16 +30,17 @@ type Scanner interface {
 	Unread()
 }
 
-// NewAST parses Flux query and produces an ast.Program.
-func NewAST(src []byte) *ast.Program {
-	f := token.NewFile("", len(src))
+// ParseFile parses Flux source and produces an ast.File.
+func ParseFile(f *token.File, src []byte) *ast.File {
 	p := &parser{
 		s: &scannerSkipComments{
 			Scanner: scanner.New(f, src),
 		},
 		src: src,
 	}
-	return p.parseProgram()
+	file := p.parseFile()
```

Comment:
Can you pass the filename as a parameter in `parseFile`? That's the general pattern that I've kept with the parser so far.